### DISCUSSION TITLE
Allow students to share a PL link to a question (that other students can see)?

### DISCUSSION THREAD

**Discussion by firasm** [PrairieLearn/PrairieLearn#6157] [Ideas]
In planning for my course this fall, I am trying to think about a way to run my Discussion boards where students ask each other for help on PL Homework questions (each student has different randomization values).

If a student tries to share a link to their question on the forum asking for help, I understand that another student shouldn't be able to see that same question (with the same numbers), however, could they be redirected to their own instance of that question (assuming it's the same question with different numbers)?

I agree this is complicated because depending on how randomization is done, the question could look completely different, but I wanted to ask if there was a way other people run their discussion forums?

What I did in the past is to ask them to use the question number and a screenshot. But this got very chaotic, very fast.

**update**: The real problem is that there’s some friction when students try to help other students. I’m trying to think about it from their perspective:
- student asks for help and posts a screenshot of their question.
- another student tries to help, but wants to make sure they did the Q correctly (it’s been a few days since they tried it).
    - **Ideally**: they click a link and go directly to their question, try a new variant, and then go back to the post and give some advice
    - **Current**: they open Canvas, click on PL, go to the assessment, then go to the question, flick back to the tab with the post to re-read the question, and then answer.
    
    
Is that too many steps for the average student? I don’t know - maybe? I don’t want students to give up if it’s not easy to help others.

**Comment by mwest1066** [comment]
I can think of at least two ways this could work:
1. We add a "share" button on questions which generates a "sharing link" which PL then redirects as appropriate for each student. This could generate a link like `/pl/course_instance/:ci_id/assessment/:a_id/assessment_question/:aq_id/` and it would redirect a student to their copy of that assessment question on that assessment. I'm thinking about Stack Overflow's share button:
    <img width="358" alt="image" src="https://user-images.githubusercontent.com/875833/186165961-d18178d2-0ef0-42ad-ae16-eb78163c319f.png">
2. We try and detect when a student is using a question URL from a different student and redirect them to one of their own variants of that question. This could be tricky because we'd also want to detect whether the current user is an instructor with student data view permissions for the current question, and not redirect them (so that they can view exact student question instances).

All that being said, for many years we've used discussion boards such as Piazza and Campuswire in large classes (many hundreds of students) and we haven't run into problems with students posting question numbers when asking for help. Screenshots are also common when the student wants to reference exact values. What "chaotic" problem are you seeing?

**Reply by firasm** [reply]
Thanks for thinking about this... I think I like option 1 the best if this is useful for other people as well.

Here's a screenshot from my discussion forum outlining the problem, there are often half a dozen to a dozen posts about the same question (at least all of them have screenshots).

<img width="472" alt="duplicates" src="https://user-images.githubusercontent.com/2507459/186196037-e0c8c930-a53a-4303-83d9-a71c1a222287.png">

I'm sure some part of this is my fault for not being stricter about how students should ask for help (this is the part I'm trying to fix this year and where the idea of sharing a link their question comes from).

On this discussion forum I can create "Issue Templates" where I ask students for various bits of information. Here's my current plan:

<img width="888" alt="Screen Shot 2022-08-23 at 8 21 56 AM" src="https://user-images.githubusercontent.com/2507459/186197754-9e529fc6-4ca7-4fa5-8cc1-e2a1b85244c9.png">

**Reply by mwest1066** [reply]
I'm on board for option (1) -- I think this would be helpful, but I'm a bit worried that it might lead to students making even lower-effort posts like "I'm stuck on <question_url>". Your template is a good idea to avoid this.
